extension/persistence/eclipselink/src/main/java/io/polaris/extension/persistence/impl/eclipselink/PolarisEclipseLinkStore.java

```diff
@@ -409,4 +413,58 @@ void deletePrincipalSecrets(EntityManager session, String clientId) {
 
     session.remove(modelPrincipalSecrets);
   }
+
+  boolean locationOverlapsWithExistingTableLike(EntityManager session, String location) {
+    if (location == null) {
+      return false;
+    }
+    diagnosticServices.check(session != null, "session_is_null");
+
+    List<String> directoryList = new ArrayList<>();
+    String queryString = """
+        SELECT
+            location
+        FROM
+            ModelEntityActive
+        WHERE
+            location IS NOT NULL
+            AND typeCode = :table_code
+            AND (
+                location LIKE CONCAT(:location, '%')
+                OR :location LIKE CONCAT(location, '%')
+            )
+    """;
+
+    int MAX_DIRECTORIES_IN_QUERY = 32;
+    Optional<List<String>> directories = PolarisUtils.pathToDirectories(location, MAX_DIRECTORIES_IN_QUERY);
+    if (directories.isPresent()) {
+      directoryList = directories.get();
+      queryString = IntStream.range(0, directoryList.size())
+          .mapToObj(i ->
+              "SELECT location " +
+              "FROM ModelEntityActive " +
+              "WHERE location IS NOT NULL " +
+              "AND typeCode = :table_code " +
+              "AND location = :directory_" + i
+          )
+          .collect(Collectors.joining(" UNION ALL "));
+
+      queryString += " UNION ALL " +
+                     "SELECT location " +
+                     "FROM ModelEntityActive " +
+                     "WHERE location IS NOT NULL " +
+                     "AND typeCode = :table_code " +
+                     "AND location LIKE :locationPrefix";
+    }
```

Comment:
Can we not do a `location IN (:directory_list)` query? I think EclipseLink must support lists as parameters.